Null pointer when remote address is null

The Request.getRemote() method throws a null pointer when context.channel().remoteAddress() returns null. The remote address can be null for various reasons.

https://github.com/zalando/logbook/blob/main/logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java#L57

## Description

## Expected Behavior
Logging layer shouldn't throw even if channel has been closed, or is otherwise wonky.

## Actual Behavior
```
Caused by: java.lang.NullPointerException: null
	at org.zalando.logbook.netty.Request.getRemote(Request.java:58)
	at org.zalando.logbook.ForwardingHttpRequest.getRemote(ForwardingHttpRequest.java:18)
	at org.zalando.logbook.ForwardingHttpRequest.getRemote(ForwardingHttpRequest.java:18)
	at org.zalando.logbook.CommonsLogFormatSink.write(CommonsLogFormatSink.java:70)
	at org.zalando.logbook.Strategy.write(Strategy.java:119)
	at org.zalando.logbook.DefaultLogbook$1.lambda$process$0(DefaultLogbook.java:55)
	at org.zalando.fauxpas.ThrowingRunnable.run(ThrowingRunnable.java:17)
	at org.zalando.logbook.netty.Sequence.runEagerly(Sequence.java:34)
	at org.zalando.logbook.netty.Sequence.set(Sequence.java:22)
	at org.zalando.logbook.netty.LogbookClientHandler.lambda$channelRead$5(LogbookClientHandler.java:71)
	at org.zalando.fauxpas.ThrowingConsumer.accept(ThrowingConsumer.java:19)
	at org.zalando.logbook.netty.Conditionals.runIf(Conditionals.java:17)
	at org.zalando.logbook.netty.LogbookClientHandler.channelRead(LogbookClientHandler.java:70)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
```


## Possible Fix
Convert null into "null", or something.

## Steps to Reproduce
Discovered issue during integration testing that involved use of WireMock. Not sure what all the ingredients are to make this happen.

## Context
Issue is causing tests to fail.

## Your Environment
* Version used: 2.14.0
* Link to your project:
* OS: MacOS

--- a/logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java
+++ b/logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java
@@ -53,8 +53,12 @@ final class Request implements org.zalando.logbook.HttpRequest, HeaderSupport {
     }
 
     @Override
+    @Nullable
     public String getRemote() {
-        return context.channel().remoteAddress().toString();
+        // According to io.netty.channel.Channel documentation,
+        // remoteAddress() returns null if the channel is not connected.
+        SocketAddress remoteAddress = context.channel().remoteAddress();
+        return (remoteAddress == null) ? null : remoteAddress.toString();
     }
 
     @Override

--- a/logbook-netty/src/test/java/org/zalando/logbook/netty/RequestUnitTest.java
+++ b/logbook-netty/src/test/java/org/zalando/logbook/netty/RequestUnitTest.java
@@ -91,10 +91,23 @@ public class RequestUnitTest {
         Request remoteRequest = new Request(context, REMOTE, req);
 
         assertThat(remoteRequest.getRequestUri()).isEqualTo("https://unknown/libs/dam/merge/metadata.json;\n" +
-                                                                ".json?path=<h1>Rhack&;%0A.inc.js");
+                ".json?path=<h1>Rhack&;%0A.inc.js");
         assertThat(remoteRequest.getPath()).isEqualTo("/libs/dam/merge/metadata.json;\n" +
-                                                          ".json");
+                ".json");
         assertThat(remoteRequest.getQuery()).isEqualTo("path=<h1>Rhack&;%0A.inc.js");
+    }
+
+    @Test
+    void shouldHandleNullRemoteAddress() {
+        HttpRequest req = mock(HttpRequest.class);
+        when(req.uri()).thenReturn("/test?a=b");
+
+        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        when(context.channel()).thenReturn(mock(Channel.class));
+        when(context.channel().remoteAddress()).thenReturn(null);
+
+        Request remoteRequest = new Request(context, REMOTE, req);
+        assertThat(remoteRequest.getRemote()).isEqualTo(null);
     }
 
     private ChannelHandlerContext mockChannelHandlerContext() {

